code/modules/mob/living/simple_animal/hostile/gremlin.dm

```diff
@@ -0,0 +1,152 @@
+//Gremlins
+//Small monsters that don't usually attack humans or other animals. Instead they mess with electronics, computers and machinery
+
+//List of objects that gremlins can't tamper with (because nobody coded an interaction for it)
+//List starts out empty. Whenever a gremlin finds a machine that it couldn't tamper with, the machine's type is added here, and all machines of such type are ignored from then on (NOT SUBTYPES)
+var/list/bad_gremlin_items = list()
+
+/mob/living/simple_animal/hostile/gremlin
+	name = "gremlin"
+	desc = "This tiny creature finds great joy in discovering and using technology. Nothing excites it more than pushing random buttons on a computer to see what it might do."
+	icon = 'icons/mob/critter.dmi'
+	icon_state = "gremlin"
+	icon_living = "gremlin"
+	icon_dead = "gremlin_dead"
+
+	health = 18
+	maxHealth = 18
+	size = SIZE_TINY
+	search_objects = 3 //Completely ignore mobs
+
+	//Tampering is handled by the 'npc_tamper()' obj proc
+	wanted_objects = list(
+		/obj/machinery,
+		/obj/structure/sink,
+		/obj/structure/displaycase
+	)
+
+	//List of objects that we don't even want to try to tamper with
+	//Subtypes of these are calculated too
+	var/list/unwanted_objects = list(/obj/machinery/atmospherics/pipe)
```

Comment:
gremlins might be more efficient if when they forget something, its type is added to unwanted objects (e.g.: once they learn they can't get anything from the light tube machinery, they won't ever try to play with the lights in a room again - which is good because there are tons of them)